elf_reader.go

```diff
@@ -413,44 +426,69 @@ func (ec *elfCode) relocateInstruction(ins *asm.Instruction, rel elf.Symbol) err
 		}
 
 	case programSection:
-		if ins.OpCode.JumpOp() != asm.Call {
-			return fmt.Errorf("not a call instruction: %v", ins)
+		if ins.OpCode.JumpOp() != asm.Call && !ins.OpCode.IsDWordLoad() {
```

Comment:
Shouldn't this check that `ins.Src == asm.PseudoFunc`? Might be worth adding a helper like `Instruction.IsLoadOfFunctionPointer()` for this, see `IsFunctionCall`.

Reply:
If I didn't miss something, in libbpf the correspondant `asm.PseudoFunc` is used to mark a `ld_imm64` that points to a subprogram and thus need relocation: https://github.com/torvalds/linux/blob/c5c17547b778975b3d83a73c8d84e8fb5ecf3ba5/tools/lib/bpf/libbpf.c#L6091
So it is up to the elf reader to check for such `ld_imm64` and then mark the instruction as a relocatable `asm.PseudoFunc`.
But I added the helper `IsLoadOfFunctionPointer` anyway, to make the code more readable in the relocation phase in `linker.go`.